src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java

```diff
@@ -191,4 +196,40 @@ public void updateAssociation(final String associationId, final Update update) {
 
     }
 
+    private AssociationDao filterForMostRecentInvitation( AssociationDao associationDao ){
+        final var invitations = associationDao.getInvitations();
+
+        if ( invitations.size() > 1 ){
+            final var mostRecentInvitation = invitations.stream()
+                    .max( Comparator.comparing( InvitationDao::getInvitedAt ) )
+                    .orElseThrow();
+
+            invitations.clear();
+            associationDao.setInvitations( List.of( mostRecentInvitation ) );
+        }
+
+        return associationDao;
+    }
+
+    @Transactional( readOnly = true )
+    public List<Invitation> fetchActiveInvitations( final String userId, final int pageIndex, final int itemsPerPage ){
+        return associationsRepository.fetchAssociationsWithActiveInvitations( userId, LocalDateTime.now() )
+                .map( this::filterForMostRecentInvitation )
+                .sorted(Comparator.comparing(AssociationDao::getApprovalExpiryAt))
+                .skip((long) pageIndex * itemsPerPage )
+                .limit( itemsPerPage )
+                .flatMap( invitationMapper::daoToDto )
+                .toList();
+    }
+
+    public List<Invitation> fetchInvitations( final AssociationDao associationDao ) {
```

Comment:
cant we just replace it with 

    public List<Invitation> fetchInvitations( final AssociationDao associationDao ) {
        return invitationMapper.daoToDto(associationDao).toList();
    }

Reply:
But then it doesn't get the `isActive` and `associationId` properties populated - the `daoToDto` this is using is a different one than the one that has that logic, it's the first one here:  

https://github.com/companieshouse/accounts-association-api/blob/0e5542212f4007ce67ee67ef8172800431952d4b/src/main/java/uk/gov/companieshouse/accounts/association/mapper/InvitationsMapper.java#L25-L33

Reply:
Oh I see yeah.  I can make the change, I misunderstood

Reply:
Got rid of the unnecessary test too since this method is simple now and the other it's using is tested: https://github.com/companieshouse/accounts-association-api/pull/101/commits/e59ef0fe44e12a7ace37559d73418d124dd6ac62 Thanks